doc/srv6/srv6_static_config_hld.md

```diff
@@ -0,0 +1,162 @@
+Besides, this document also define new YANG model specification and unit-test cases used to validate the aforementioned features.
+
+
+# 2 Feature Requirements
+
+## 2.1 Functional Requirements
+
+Provide ability to statically configure SRv6 SIDs for block IDs, locators and local functions from CONFIG_DB.
+
+## 2.2 Configuration and Management Requirements
+
+1. User should be able to statically configure block length, locator length and function length for SRv6.
+
+2. User should be able to statically configure a number of SIDs/uSIDs for the local functions of the switch.
+
+## 2.3 Warm Boot Requirements
+
+Warm reboot is intended to be supported for planned system warm reboot.
+
+ 
+
+# 3 Feature Design
+
+At the time of writing this document, FRR has been able to program the SRv6 related tables in APPL_DB through fpmsyncd.
+However, there is still one gap preventing SONiC being utilized for SRv6 SDN deployment.
+Specifically, there is no mechamism in SONiC allowing SDN controllers or users to directly add configuration for SRv6 without involving BGP.
+
+In this document, we first define a new **SRV6_MY_SID_TABLE** table in CONFIG_DB that serves as the configuration source of SRv6 in SONiC.
+Then, we design a new SRv6 Manager module in bgpcfgd to subscribe to the **SRV6_MY_SID_TABLE** table and compile changes in CONFIG_DB to changes in the configurations of FRR.
+To verify the correctness of the aforementioned flow, we also define the relevant YANG model specification.
+The workflow of the new mechanism is shown in the following diagram.
+
+![Static SRv6 Config flow](images/SRv6_bgpcfgd.png)
+
+The design details of each step is described in the following subsections.
+
+## 3.1 New Table in ConfigDB
+
+
+**SRV6_MY_SID_TABLE**
+
+Description: New table to hold local SID definition and SID to behavior mapping. (A simplified redefinition of SRV6_MY_SID_TABLE in [SRv6_HLD](./srv6_hld.md))
+
+Schema:
+
+```
+; New table
+; holds local SID to behavior mapping, allow 1:1 or n:1 mapping
+
+key = SRV6_MY_SID_TABLE|ipv6address
+; field = value
+block_len = blen             ; bit length of block portion in address, default 32
+node_len = nlen              ; bit length of node ID portion in address, default 16
+func_len = flen              ; bit length of function portion in address, default 16
+arg_len = alen               ; bit length of argument portion in address, default 0
+action = behavior            ; behaviors defined for the SID, default uN
+vrf = VRF_TABLE.key          ; Optional, VRF name for decapsulation actions, only applicable to "uDT6" and "uDT46" by now, default "default"
+
+For example:
+    "SRV6_MY_SID_TABLE" : {
+        "FCBB:BBBB:20::" : {
+           "action": "uN"
+        },
+        "FCBB:BBBB:20:F1::" : {
+           "action": "uDT46",
+        },
+        "FCBB:BBBB:20:F2::" : {
+           "action": "uDT46",
+           "vrf":  "VRF-1001"
+        },
+    }
+```
+
+
+## 3.2 Bgpcfgd changes
+
```

Comment:
Can you please capture the FRR configuration sample. It would be good if you can provide sonic field to FRR configuration mapping.

Reply:
There is still some ongoing discussion on the details of the new FRR configuration CLI. We'd better add the config sample and mapping later.